tests/python_package_test/test_sklearn.py

```diff
@@ -445,9 +445,19 @@ def test_clone_and_property():
     gbm.fit(X, y)
 
     gbm_clone = clone(gbm)
+
+    # original estimator is unaffected
```

Comment:
via this warning...

> ./tests/python_package_test/test_sklearn.py:447:5: F841 local variable 'gbm_clone' is assigned to but never used

...`flake8` caught that this test wasn't really checking the correctness of `sklearn.clone()` applied to an `LGBMRegressor`. It was basically just checking that `clone()` ran without error.

I chose to fix the warning by also just making the test stricter.

See https://scikit-learn.org/stable/modules/generated/sklearn.base.clone.html#sklearn.base.clone for details on what this function does.